core/transfer/registry/registry.go

```diff
@@ -42,37 +43,73 @@ func init() {
 	plugins.Register(&transfertypes.OCIRegistry{}, &OCIRegistry{})
 }
 
+type registryOpts struct {
```

Comment:
Any particular reason to leave this unexported, given its part of an exported type?

Reply:
@cpuguy83 It just follows the pattern other parts of containerd takes (e.g. `core/remotes/docker`). If exporting the struct is preferred, I can fix the code to export it.

https://github.com/containerd/containerd/blob/93022d83d14ace66f1d0c225f2f9993c6a2eb20c/core/remotes/docker/registry.go#L113-L156

Reply:
ping @cpuguy83
Any use cases for exported `registryOpts`?

Reply:
This is the normal pattern we use since having custom "With*" functions doesn't make sense on these types.

Reply:
Makes sense.
Sorry for the delayed response.